src/miniscript/astelem.rs

```diff
@@ -633,7 +633,7 @@ impl StackCtxOperations for script::Builder {
         let mut builder = self;
         // Initial Witness
         let max_elems = MAX_SCRIPT_ELEMENT_SIZE / MAX_STANDARD_P2WSH_STACK_ITEM_SIZE;
-        for _ in 0..(max_elems - 1) {
+        for _ in 0..max_elems {
```

Comment:
This took me a bit of thinking :) I think that `max_elems` is misleadingly named because it's actually one less than the maximum number of elements. I think it'd be better to leave the `- 1` in place (no parentheses are needed btw) and to change

```
        let max_elems = MAX_SCRIPT_ELEMENT_SIZE / MAX_STANDARD_P2WSH_STACK_ITEM_SIZE;
￼```
to
```
        let max_elems = (MAX_SCRIPT_ELEMENT_SIZE + MAX_STANDARD_P2WSH_STACK_ITEM_SIZE - 1) / MAX_STANDARD_P2WSH_STACK_ITEM_SIZE;
￼```
or even just
```
        let max_elems = MAX_SCRIPT_ELEMENT_SIZE / MAX_STANDARD_P2WSH_STACK_ITEM_SIZE + 1;
```
since these numbers are fixed and we know that one is not a multiple of the other.